tests/UnifiedSpecTests/UnifiedTestRunner.php

```diff
@@ -213,93 +215,41 @@ private function doTestCase(stdClass $test, string $schemaVersion, array $runOnR
     /**
      * Checks server version and topology requirements.
      *
+     * Arguments for RunOnRequirement::isSatisfied() will be cached internally.
+     *
      * @throws SkippedTest unless one or more runOnRequirements are met
      */
     private function checkRunOnRequirements(array $runOnRequirements)
     {
+        static $cachedIsSatisfiedArgs;
+
         assertNotEmpty($runOnRequirements);
         assertContainsOnly('object', $runOnRequirements);
 
-        $serverVersion = $this->getCachedServerVersion();
-        $topology = $this->getCachedTopology();
-        $serverParameters = $this->getCachedServerParameters();
+        if (! isset($cachedIsSatisfiedArgs)) {
+            $cachedIsSatisfiedArgs = [
+                $this->getServerVersion(),
+                $this->getTopology(),
+                $this->getServerParameters(),
+                $this->isAuthenticated(),
+                $this->isServerless(),
+            ];
+        }
```

Comment:
Once I started adding new fields to the requirements, it made a lot more sense to cache them in one place (vs. the two methods we previously had).